src/TuDa.CIMS.Web/Services/IWorkingGroupApi.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Text.Json;
+using Refit;
+using TuDa.CIMS.Shared.Attributes.ServiceRegistration;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+using TuDa.CIMS.Web.Extensions;
+
+namespace TuDa.CIMS.Web.Services;
+
+/// <summary>
+/// Refit client interface for performing operations on WorkingGroups.
+/// </summary>
+[RefitClient("/api/working-group")]
+public interface IWorkingGroupApi
+{
+    /// <summary>
+    /// Retrieves a Working Group by its unique identifier.
+    /// </summary>
+    /// <param name="id">The unique identifier of the Working Group.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="ErrorOr{T}"/> containing the Working Group if successful.
+    /// </returns>
+    public async Task<ErrorOr<WorkingGroup>> GetAsync(Guid id) =>
+        await GetAsyncInternal(id).ToErrorOrAsync();
+
+
+    /// <summary>
+    /// Internal method to retrieve a Working Group by its unique identifier.
+    /// </summary>
+    /// <param name="id">The unique identifier of the Working Group.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="IApiResponse{T}"/> containing the WorkingGroup.
+    /// </returns>
+    /// <remarks>This will be generated by <see cref="Refit"/>.</remarks>
+    [Get("/{id}")]
+    protected Task<IApiResponse<WorkingGroup>> GetAsyncInternal(Guid id);
+
+    /// <summary>
+    /// Retrieves all Working Groups.
+    /// </summary>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="ErrorOr{T}"/> containing an IEnumerable of Working Groups if successful.
+    /// </returns>
+    public async Task<ErrorOr<IEnumerable<WorkingGroup>>> GetAllAsync() =>
+        await GetAllAsyncInternal().ToErrorOrAsync();
+
+    /// <summary>
+    /// Internal method to retrieve all Working Groups.
+    /// </summary>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="IApiResponse{T}"/> containing an IEnumerable of Working Groups.
+    /// </returns>
+    /// <remarks>This will be generated by <see cref="Refit"/>.</remarks>
+    [Get("/")]
+    protected Task<IApiResponse<IEnumerable<WorkingGroup>>> GetAllAsyncInternal();
+
+    /// <summary>
+    /// Creates a new Working Group.
+    /// </summary>
+    /// <param name="item">The Working Group to create.</param>
+    /// <returns>
+    /// A Task representing the asynchronous operation, with a result of <see cref="ErrorOr{T}"/> containing the unique identifier of the created Working Group if successful.
+    /// </returns>
+    /// <remarks>
+    /// Uses <see cref="JsonSerializer"/> to ensure proper serialization of polymorphic types.
+    /// </remarks>
+    public async Task<ErrorOr<Guid>> CreateAsync(WorkingGroup item) =>
```

Comment:
🔧 The parameter needs to be of type `UpdateWorkingGroupDto`

Reply:
not CreateWorkingGroupDto?

Reply:
Yeah, you're right 😅